src/seedu/addressbook/commands/Command.java

```diff
@@ -42,6 +42,8 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
     public CommandResult execute(){
         throw new UnsupportedOperationException("This method should be implement in child classes");
     }
+    
+    public abstract boolean isMutating();
```

Comment:
This method should have a header comment. If not, users of the method as well as those who overrides method will not know how exactly the method behaves. e.g. `Returns true if …`